packages/integration-tests/src/test-project-setup/conversation_handler_project.ts

```diff
@@ -158,7 +210,130 @@ class ConversationHandlerTestProject extends TestProjectBase {
         },
       ],
       request: {
-        headers: { authorization: authenticatedUserCredentials.accessToken },
+        headers: { authorization: accessToken },
+      },
+      responseMutationInputTypeName:
+        'CreateConversationMessageAssistantResponseInput',
+      responseMutationName: 'createConversationMessageAssistantResponse',
+      modelConfiguration: {
+        modelId: 'anthropic.claude-3-haiku-20240307-v1:0',
+        systemPrompt: 'You are helpful bot.',
+      },
+    };
+    await this.executeConversationTurn(
+      event,
+      defaultConversationHandlerFunction,
+      apolloClient
+    );
+  };
+
+  private assertDefaultConversationHandlerCanExecuteTurnWithDataTool = async (
+    backendId: BackendIdentifier,
+    accessToken: string,
+    graphqlApiEndpoint: string,
+    apolloClient: ApolloClient<NormalizedCacheObject>
+  ): Promise<void> => {
+    const defaultConversationHandlerFunction = (
+      await this.resourceFinder.findByBackendIdentifier(
+        backendId,
+        'AWS::Lambda::Function',
+        (name) => name.includes('default')
+      )
+    )[0];
+
+    // send event
+    const event: ConversationTurnEvent = {
+      conversationId: randomUUID().toString(),
+      currentMessageId: randomUUID().toString(),
+      graphqlApiEndpoint: graphqlApiEndpoint,
+      messages: [
+        {
+          role: 'user',
+          content: [
+            {
+              text: 'What is the temperature in Seattle?',
+            },
+          ],
+        },
+      ],
+      request: {
+        headers: { authorization: accessToken },
+      },
+      responseMutationInputTypeName:
+        'CreateConversationMessageAssistantResponseInput',
+      responseMutationName: 'createConversationMessageAssistantResponse',
+      modelConfiguration: {
+        modelId: 'anthropic.claude-3-haiku-20240307-v1:0',
+        systemPrompt: 'You are helpful bot.',
+      },
+      toolsConfiguration: {
+        tools: [
+          {
+            name: 'thermometer',
+            description: 'Provides the current temperature for a given city.',
+            inputSchema: {
+              json: {
+                type: 'object',
+                properties: {
+                  city: {
+                    type: 'string',
+                    description: 'string',
+                  },
+                },
+                required: [],
+              },
+            },
+            graphqlRequestInputDescriptor: {
+              queryName: 'getTemperature',
+              selectionSet: ['value', 'unit'],
+              propertyTypes: {
+                city: 'String',
+              },
+            },
+          },
+        ],
+      },
+    };
+    const response = await this.executeConversationTurn(
+      event,
+      defaultConversationHandlerFunction,
+      apolloClient
+    );
+    // Assert that tool was used.
+    assert.match(response.content, /85/);
```

Comment:
85 is the expected temperature?

Reply:
should we also match the unit here? for unit in the schema 'F'

Reply:
correct.

Reply:
> should we also match the unit here? for unit in the schema 'F'

I was trying to but LLM converted it to some sentence with `Fahrenheit` .
The point of this test is to assure that tool was ran and ran successfully. The numerical value seems to be good enough signal that this has happened. But given how LLM work too strict assertion here will be problematic.

Reply:
Oh, that is interesting. So we are asserting on what can be guaranteed, i'll keep that in mind; is the LLM always going to respond with numeric 75 - versus seventy-five ?

Reply:
> is the LLM always going to respond with numeric 75 - versus seventy-five ?

Hard to say. But for now it seems to be best bet. Hopefully behavior is somewhat fixed by hardcoding modelid (it is versioned). However, while playing with other questions I did see some variations of answers.